pkg/controller/gather_commands.go

```diff
@@ -409,11 +410,8 @@ func wasDataProcessed(ctx context.Context,
 	delay := conf.DataReporting.ReportPullingDelay
 	klog.Infof("Initial delay when checking processing status: %v", delay)
 
-	// requestRetryCounter is used to retry an http request when the response != 200
-	requestRetryCounter := 0
-	// networkRetryCounter is used to retry when there is a network issue that caused failure
 	networkRetryCounter := 0
-	// statusRetryCounter is used to retry when the processing pipeline did not finished yet
+	requestRetryCounter := 0
 	statusRetryCounter := 0
```

Comment:
I think we could wrap these counter in some struct for example like this (maybe the naming could be improved)
```suggestion
	type retryCounter struct {
		networkRetryCounter int
		requestRetryCounter int
		statusRetryCounter int

		masRetryCount int
	}
```
Then we could initialize it with a limit like this, which could also simplify unit testing
```suggestion
	rc := &retryCounter{masRetryCount: numberOfStatusQueryRetries}
```